app/controllers/concerns/saml_idp_auth_concern.rb

```diff
@@ -237,7 +237,12 @@ def saml_response_signature_options
 
   def saml_request_service_provider
     return @saml_request_service_provider if defined?(@saml_request_service_provider)
-    @saml_request_service_provider = ServiceProvider.find_by(issuer: current_issuer)
+    @saml_request_service_provider =
+      if current_issuer.blank?
+        nil
+      else
+        ServiceProvider.find_by(issuer: current_issuer)
+      end
```

Comment:
Technically we could make this a single `if` without the `else` and have this assign as `nil` with working memoization, though I think maybe this a good instance of being overly explicit due to how confusing the conditional assignment can be in Ruby.

```rb
    @saml_request_service_provider =
      if current_issuer.present?
        ServiceProvider.find_by(issuer: current_issuer)
      end
```

Reply:
Ah, you're right, that hadn't occurred to me. I kind of prefer the explicitness though.